src/snark_intf.ml

```diff
@@ -549,29 +563,29 @@ let multiply3 (x : Field.Var.t) (y : Field.Var.t) (z : Field.Var.t)
 
     val run_unchecked :
          public_input:(unit, 'public_input) H_list.t
-      -> ?handler:Request.Handler.t
+      -> ?handlers:Handler.t list
```

Comment:
I think this could just be a function `Request.t -> Response.t` -- what do you think?

Reply:
I like the idea of it being a list, so you can pass several separately-defined handlers without having to jerry-rig your own `Request.Handler.run` to stitch them together. Switching from `Handler.t` to `Request.t -> Request.Response.t` makes sense to me though.

Reply:
@imeckler I've just had a proper look at this, seems like we can't make the change because we need the record field polymorphism from `Request.request`. I think the only real change we could make on this is dropping the `list` (or not).